### PR TITLE
Add FXIOS-7168 [v119] Credential Autofill coordinator

### PR DESCRIPTION
**Pull request by FilippoZazzeroni**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7168)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15935)

## :bulb: Description
Add `CredentialAutofillCoordinator` and its related nimbus feature.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [x] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by OrlaM** [comment]
I think the fact the view is a bottom sheet card is an implementation detail of the flow so I don't think everything should be named using this terminology. Everything seems to pertain to the credit card autofill flow so maybe something like CreditCardAutofillCoordinator? Or maybe even CredentialAutofillCoordinator to leave room for future features like address autofill or password that will have a similar flow.

**Comment by FilippoZazzeroni** [comment]
@OrlaM ok sounds good, indeed the bottom sheet card si too generic, and doesn't represent the actual flow. I will name it credit card auto fill coordinator

File: Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    func show(homepanelSection: Route.HomepanelSection) {
+        showHomepanelSectionCalled += 1
+    }
+
+    func showFakespotFlow(productURL: URL) {
+        showFakespotCalled += 1
+    }
+
+    func showEnhancedTrackingProtection(sourceView: UIView) {
+        showEnchanedTrackingProtectionCalled += 1
```
Comment (lmarceau) [inline Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift:46]: typo
```suggestion
        showEnhancedTrackingProtectionCalled += 1
```

File: Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+@testable import Client
+
+class MockBrowserCoordinator: BrowserNavigationHandler, ParentCoordinatorDelegate {
+    var showSettingsCalled = 0
+    var showFakespotCalled = 0
+    var showCreditCardAutofillCalled = 0
+    var showRequiredPassCodeCalled = 0
+    var showLibraryCalled = 0
+    var showHomepanelSectionCalled = 0
+    var showEnchanedTrackingProtectionCalled = 0
```
Comment (lmarceau) [inline Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift:17]:
```suggestion
    var showEnhancedTrackingProtectionCalled = 0
```

**Review by lmarceau — COMMENTED** [review]
Small nit only!

File: Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift (hunk 2 above)
Comment (FilippoZazzeroni) [reply Tests/ClientTests/Coordinators/Mocks/MockBrowserCoordinator.swift:17]: @lmarceau thank you. i'll modify it just now

**Review by lmarceau — APPROVED** [review]

**Pull request merged**